src/Catalog/Dnx/DnxCatalogCollector.cs

```diff
@@ -21,122 +23,182 @@ public class DnxCatalogCollector : CommitCollector
         private readonly StorageFactory _storageFactory;
         private readonly DnxMaker _dnxMaker;
         private readonly ILogger _logger;
+        private readonly int _maxDegreeOfParallelism;
 
         public DnxCatalogCollector(
             Uri index,
             StorageFactory storageFactory,
             ITelemetryService telemetryService,
             ILogger logger,
+            int maxDegreeOfParallelism,
             Func<HttpMessageHandler> handlerFunc = null,
             TimeSpan? httpClientTimeout = null)
             : base(index, telemetryService, handlerFunc, httpClientTimeout)
         {
             _storageFactory = storageFactory;
             _dnxMaker = new DnxMaker(storageFactory);
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _maxDegreeOfParallelism = maxDegreeOfParallelism;
         }
 
         public Uri ContentBaseAddress { get; set; }
 
-        protected override async Task<bool> OnProcessBatch(CollectorHttpClient client, IEnumerable<JToken> items, JToken context, DateTime commitTimeStamp, bool isLastBatch, CancellationToken cancellationToken)
+        protected override async Task<bool> OnProcessBatch(
+            CollectorHttpClient client,
+            IEnumerable<JToken> items,
+            JToken context,
+            DateTime commitTimeStamp,
+            bool isLastBatch,
+            CancellationToken cancellationToken)
+        {
+            var catalogEntries = items.Select(item =>
```

Comment:
This LINQ query is going to be run once for `AssertNoMultipleEntriesForSamePackageIdentity` and once for `ProcessCatalogEntriesAsync`. Should it be executed once with `.ToList()` instead?